### ISSUE TITLE
"The guest machine entered an invalid state" in a MacOS M1 with VirtualBox

### ISSUE THREAD

**Issue by nbro10** [hashicorp/vagrant#13098]
### Debug output

Here are the steps.

1. I installed VirtualBox [Developer preview for macOS / Arm64 (M1/M2) hosts](https://download.virtualbox.org/virtualbox/7.0.6/VirtualBox-7.0.6_BETA4-155176-macOSArm64.dmg)
2. `brew install hashicorp/tap/hashicorp-vagrant`
3. `vagrant autocomplete install --bash --zsh`
4. `vagrant init hashicorp/bionic64`
5. `vagrant up`

This is the result

```
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'hashicorp/bionic64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'hashicorp/bionic64' version '1.0.282' is up to date...
==> default: Setting the name of the VM: vagrant_experiments_default_1677497868687_62809
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection reset. Retrying...
The guest machine entered an invalid state while waiting for it
to boot. Valid states are 'starting, running'. The machine is in the
'aborted' state. Please verify everything is configured
properly and try again.

If the provider you're using has a GUI that comes with it,
it is often helpful to open that and watch the machine, since the
GUI often has more helpful error messages than Vagrant can retrieve.
For example, if you're using VirtualBox, run `vagrant up` while the
VirtualBox GUI is open.

The primary issue for this error is that the provider you're using
is not properly configured. This is very rarely a Vagrant issue.
```

I did not modify the vagrant file or any configuration of VirtualBox.

### Expected behavior

No error and I can boot the VM.

### Actual behavior

See logs above

### Reproduction information 

See above

#### Vagrant version

`Vagrant 2.3.4` 

#### Host operating system

MacOS 

- m1, 2020
- Monterey (12.5.1)

#### Guest operating system

`hashicorp/bionic64`

#### Steps to reproduce

See above

#### Vagrantfile

```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

# All Vagrant configuration is done below. The "2" in Vagrant.configure
# configures the configuration version (we support older styles for
# backwards compatibility). Please don't change it unless you know what
# you're doing.
Vagrant.configure("2") do |config|
  # The most common configuration options are documented and commented below.
  # For a complete reference, please see the online documentation at
  # https://docs.vagrantup.com.

  # Every Vagrant development environment requires a box. You can search for
  # boxes at https://vagrantcloud.com/search.
  config.vm.box = "hashicorp/bionic64"

  # Disable automatic box update checking. If you disable this, then
  # boxes will only be checked for updates when the user runs
  # `vagrant box outdated`. This is not recommended.
  # config.vm.box_check_update = false

  # Create a forwarded port mapping which allows access to a specific port
  # within the machine from a port on the host machine. In the example below,
  # accessing "localhost:8080" will access port 80 on the guest machine.
  # NOTE: This will enable public access to the opened port
  # config.vm.network "forwarded_port", guest: 80, host: 8080

  # Create a forwarded port mapping which allows access to a specific port
  # within the machine from a port on the host machine and only allow access
  # via 127.0.0.1 to disable public access
  # config.vm.network "forwarded_port", guest: 80, host: 8080, host_ip: "127.0.0.1"

  # Create a private network, which allows host-only access to the machine
  # using a specific IP.
  # config.vm.network "private_network", ip: "192.168.33.10"

  # Create a public network, which generally matched to bridged network.
  # Bridged networks make the machine appear as another physical device on
  # your network.
  # config.vm.network "public_network"

  # Share an additional folder to the guest VM. The first argument is
  # the path on the host to the actual folder. The second argument is
  # the path on the guest to mount the folder. And the optional third
  # argument is a set of non-required options.
  # config.vm.synced_folder "../data", "/vagrant_data"

  # Provider-specific configuration so you can fine-tune various
  # backing providers for Vagrant. These expose provider-specific options.
  # Example for VirtualBox:
  #
  # config.vm.provider "virtualbox" do |vb|
  #   # Display the VirtualBox GUI when booting the machine
  #   vb.gui = true
  #
  #   # Customize the amount of memory on the VM:
  #   vb.memory = "1024"
  # end
  #
  # View the documentation for the provider you are using for more
  # information on available options.

  # Enable provisioning with a shell script. Additional provisioners such as
  # Ansible, Chef, Docker, Puppet and Salt are also available. Please see the
  # documentation for more information about their specific syntax and use.
  # config.vm.provision "shell", inline: <<-SHELL
  #   apt-get update
  #   apt-get install -y apache2
  # SHELL
end
```

### Other

There are similar issues:

- https://github.com/hashicorp/vagrant/issues/12793
- https://github.com/hashicorp/vagrant/issues/2157

But they don't seem to be exactly the same

I don't know if I need to change anything in the VirtualBox, vagrant file or anywhere else. This is my first time using vagrant after a long time and my first time using vagrant in an m1.

It's possible that this is a VirtualBox issue or misconfiguration. I've tried to run the box from the VirtualBox GUI, waited some time, then a menu to select Ubuntu appeared and I selected that, then waited some more time, and the VirtualBox crashed...  so far so good 😞 

Here are the windows that pop up after the crash

![crash1](https://user-images.githubusercontent.com/106664639/221559417-25ea54ce-165f-411e-ae65-91aa76c22050.png)

![crash2](https://user-images.githubusercontent.com/106664639/221559444-fa82c96e-ed07-47a8-8250-9b237e4f4ae8.png)

### Related theads

- https://forums.virtualbox.org/viewtopic.php?f=8&t=107959
- https://forums.virtualbox.org/viewtopic.php?f=8&t=107871
- https://forums.virtualbox.org/viewtopic.php?f=8&t=108428

**Comment by phinze** [comment]
Hi @nbro10 - thank you for the detailed report!

The first thing I notice here is that you're using `hashicorp/bionic64` which only contains x86 boxes. You need to use an ARM64 box to get it to work on an M1.

One ARM64 box I found to try that has a virtualbox provider would be [this one](https://app.vagrantup.com/ilker/boxes/ubuntu2004) (note this is not an official box so keep that in mind).

Let me know switching to an ARM box resolves the issues you're seeing!

**Comment by douglasawh** [comment]
I have the same issue but my Mac is different: 
2021
13.2.1 (22D68) 

Also, I used the suggested image:

==> default: Successfully added box 'ilker/ubuntu2004' (v1.0) for 'virtualbox'!

**Comment by bmuna** [comment]
hi @nbro10, just run into this, did you find a way to fix the error, 
chang the box to 'ilker/ubuntu2004' didn't work for me

**Comment by engelmohr** [comment]
Hey @nbro10 this seems to be a problem with early stage of Virtualbox for ARM support as discussed here: https://forums.virtualbox.org/viewtopic.php?f=8&t=107344

Essentially the current "BETA" of Virtualbox for ARM is just an alpha version at best. Or to say in layman's terms it is basically useless as of now. I suggest to user another provider.

**Comment by oluchristian** [comment]
I am having same issue with my M1

**Comment by chrisroberts** [comment]
Hi there,

As described in [this](https://forums.virtualbox.org/viewtopic.php?f=8&t=107344) thread linked to by @engelmohr above, VirtualBox is generally not functional on ARM processors. Based on the description of current functionality in the linked thread, it can run a 32-bit single core x86 guest (very slowly) and that will likely not cover the majority of use cases.

Cheers!
